geoapps/inversion/line_sweep/driver.py

```diff
@@ -186,16 +199,16 @@ def collect_results(self):
 
         octree_model.copy(parent=self.pseudo3d_params.out_group)
 
-    def collect_line_data(self, survey, data):
+    def collect_line_data(self, survey, line_indices, data):
+        """
+        Fill chunks of values from one line
+        """
         for child in survey.children:  # initialize data values dictionary
             if "Iteration" in child.name and child.name not in data:
-                data[child.name] = {"values": np.zeros(survey.n_cells)}
+                data[child.name] = {"values": np.zeros_like(line_indices) * np.nan}
 
-        ind = None
-        for child in survey.children:  # fill a chunk of values from one line
+        for child in survey.children:
             if "Iteration" in child.name:
-                if ind is None:
-                    ind = ~np.isnan(child.values)
-                data[child.name]["values"][ind] = child.values[ind]
+                data[child.name]["values"][line_indices] = child.values
```

Comment:
won't this explode the number of entities?  Seems like you'll get an Iteration 0, 1, 2, ... for one line, then an Iteration 0 (0), 1 (0), 2(0), ... for the next line, and so on??

Reply:
I don't think so? This line would just be inserting values in the full array for every iteration it has. All lines share the same data iteration names, so the dict just get updated.

Reply:
we'll find out for sure during test!